docs/book/src/reference/versions.md

```diff
@@ -265,18 +266,18 @@ In some cases, also Cluster API and/or Cluster API providers are defining additi
 The following table defines the support matrix for the Cluster API core provider.
 See [Cluster API release support](#cluster-api-release-support) and [Kubernetes versions support](#kubernetes-versions-support).
 
-|                  | v1.8, _Maintenance Mode_ | v1.9              | v1.10             |
+|                  | v1.9, _Maintenance Mode_ | v1.10             | v1.11             |
 |------------------|--------------------------|-------------------|-------------------|
 | Kubernetes v1.24 |                          |                   |                   |
-| Kubernetes v1.25 | ✓ (only workload)        |                   |                   |
-| Kubernetes v1.26 | ✓ (only workload)        | ✓ (only workload) |                   |
-| Kubernetes v1.27 | ✓                        | ✓ (only workload) | ✓ (only workload) |
-| Kubernetes v1.28 | ✓                        | ✓                 | ✓ (only workload) |
-| Kubernetes v1.29 | ✓                        | ✓                 | ✓                 |
+| Kubernetes v1.25 |                          |                   |                   |
+| Kubernetes v1.26 | ✓ (only workload)        |                   |                   |
+| Kubernetes v1.27 | ✓ (only workload)        | ✓ (only workload) |                   |
+| Kubernetes v1.28 | ✓                        | ✓ (only workload) | ✓ (only workload) |
+| Kubernetes v1.29 | ✓                        | ✓                 | ✓ (only workload) |
 | Kubernetes v1.30 | ✓                        | ✓                 | ✓                 |
-| Kubernetes v1.31 | ✓ >= v1.8.1              | ✓                 | ✓                 |
-| Kubernetes v1.32 |                          | ✓ >= v1.9.1       | ✓                 |
-| Kubernetes v1.33 |                          |                   | ✓ >= v1.10.1      |
+| Kubernetes v1.31 | ✓                        | ✓                 | ✓                 |
+| Kubernetes v1.32 | ✓ >= v1.9.1              | ✓                 | ✓                 |
+| Kubernetes v1.33 |                          | ✓ >= v1.10.1      | ✓                 |
```

Comment:
This is not for this PR, as 1.33 support was already added for v1.10.1... but I'm curious why we aren't pinning to the same go version for 1.33 if we're introducing support for it?

Ref:

- https://github.com/kubernetes-sigs/cluster-api/blob/v1.10.1/go.mod#L3

Compare to:

- https://github.com/kubernetes/kubernetes/blob/v1.33.0/go.mod#L9

Updating to go 1.24 would obviously include updating a bunch of core libraries (apimachinery, etc), but I was under the impression that "CAPI support" for a particular version would include that affinity.

Reply:
> why we aren't pinning to the same go version for 1.33...?

It's a good point, CAPI usually keeps those in sync.

Switching to Go 1.24 is in progress: see #12128, which may be blocked by #12088.

Reply:
Small addition. CR v0.21 which would use k8s.io/* v0.33 dependencies is not released yet even as of today.

Even if it would be released slightly before the CAPI release I don't want to pick up a new CR (and also k8s.io/* & Go minor version) with basically zero soak time